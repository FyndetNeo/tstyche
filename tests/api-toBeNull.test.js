import { expect, test } from "@jest/globals";
import * as tstyche from "tstyche";
import { getFixtureUrl } from "./__utils__/fixtureFactory.js";
import { normalizeOutput } from "./__utils__/normalizeOutput.js";
import { spawnTyche } from "./__utils__/spawnTyche.js";

const fixtureUrl = getFixtureUrl("api-toBeNull");

test("'toBeNull' implementation", () => {
  expect(tstyche.expect).toHaveProperty("type.toBeNull", expect.any(Function));
});

test("toBeNull", async () => {
  const { exitCode, stderr, stdout } = await spawnTyche(fixtureUrl);

  expect(normalizeOutput(stdout)).toMatchSnapshot("stdout");
  expect(normalizeOutput(stderr)).toMatchSnapshot("stderr");

  expect(exitCode).toBe(1);
});
