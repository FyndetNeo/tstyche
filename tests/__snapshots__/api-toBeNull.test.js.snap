// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toBeNull: stderr 1`] = `
"Error: Type 'null' is not identical to type 'void'.

   7 |   expect(returnsNull()).type.toBeNull();
   8 | 
>  9 |   expect(returnsVoid()).type.toBeNull();
     |                              ^
  10 | });
  11 | 
  12 | test("is NOT null?", () => {

       at ./__typetests__/toBeNull.tst.ts:9:30 ❭ is null?

Error: Type 'null' is identical to type 'null'.

  13 |   expect(returnsVoid()).type.not.toBeNull();
  14 | 
> 15 |   expect(returnsNull()).type.not.toBeNull();
     |                                  ^
  16 | });
  17 | 

       at ./__typetests__/toBeNull.tst.ts:15:34 ❭ is NOT null?

"
`;

exports[`toBeNull: stdout 1`] = `
"uses TypeScript <<version>> with ./tsconfig.json

fail ./__typetests__/toBeNull.tst.ts
  × is null?
  × is NOT null?

Targets:    1 failed, 1 total
Test files: 1 failed, 1 total
Tests:      2 failed, 2 total
Assertions: 2 failed, 2 passed, 4 total
Duration:   <<timestamp>>

Ran all test files.
"
`;
